File: src/main/java/br/com/jiratorio/service/UserConfigService.java
package br.com.jiratorio.service;

import br.com.jiratorio.domain.ImportHolidayInfo;
import br.com.jiratorio.domain.request.UpdateUserConfigRequest;
import br.com.jiratorio.domain.response.UserConfigResponse;

public interface UserConfigService {

    void update(String username, UpdateUserConfigRequest updateUserConfigRequest);

    ImportHolidayInfo retrieveHolidayInfo(String username);

    UserConfigResponse findByUsername(String username);

}

File: src/main/kotlin/br/com/jiratorio/service/UserConfigService.kt
package br.com.jiratorio.service

import br.com.jiratorio.domain.ImportHolidayInfo
import br.com.jiratorio.domain.request.UpdateUserConfigRequest
import br.com.jiratorio.domain.response.UserConfigResponse

interface UserConfigService {

    fun update(username: String, updateUserConfigRequest: UpdateUserConfigRequest)

    fun retrieveHolidayInfo(username: String): ImportHolidayInfo

    fun findByUsername(username: String): UserConfigResponse

}
